src/i_video.c

```diff
@@ -638,6 +639,15 @@ void I_DynamicResolution(void)
         return;
     }
 
+    if (newheight < oldheight)
```

Comment:
So, we do not directly link maximum distance to rendering resolution. But when it's necessary to decrease resolution, we decrease maximum distance as well and vice versa. This means, we could end up with different maximum distances for the same resolution, depending on which direction we reached the resolution from. Right?

Reply:
> This means, we could end up with different maximum distances for the same resolution, depending on which direction we reached the resolution from. Right?

Right, but that is not a problem, I think. What I have tried to do here is link the max distance and the frame time.

Reply:
Sounds reasonable, thank you!